### PR TITLE
Slot storage

### PR DESCRIPTION
**Pull request by shenghaoc**
Currently I am saving an additional copy of the pet in each slot.
But reading and writing is fine.

### CODE REVIEW

**Comment by shenghaoc** [comment]
Originally, all the variables have a wrapper around them.
Therefore, when calling toModelType() there is a isValidXXX() check. However, dateTime and duration are Java library classes that do not have such methods.
@umaikaze, consider creating wrappers for them?

**Comment by shenghaoc** [comment]
There's very little to do for storage and it already works, only issue would be that without a wrapper class I have to manually copy the checks from AddSlot.
It's just going from storing and adding all the pets, to storing and adding all the pets and slots.
Writing tests now.

File: src/test/java/seedu/address/logic/commands/slot/EditSlotDescriptorTest.java [hunk 1]
```diff
@@ -7,17 +7,19 @@
...
 import seedu.address.logic.commands.slot.EditSlotCommand.EditSlotDescriptor;
 import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
 import seedu.address.testutil.slot.EditSlotDescriptorBuilder;
 
 class EditSlotDescriptorTest {
 
-    private Model model = getTypicalModelManager();
+    private Model model = new ModelManager(getTypicalPetTracker(), new UserPrefs());
```
Comment (shenghaoc) [inline src/test/java/seedu/address/logic/commands/slot/EditSlotDescriptorTest.java]: Need to check whether this relies on model.

File: src/test/java/seedu/address/testutil/slot/SlotBuilder.java [hunk 2]
```diff
@@ -38,12 +39,11 @@ public SlotBuilder(Slot slotToCopy) {
...
     }
 
-    // TODO Make use of the model passed in
     /**
      * Sets the {@code Pet} of the {@code Slot} that we are building.
      */
-    public SlotBuilder withPet(Pet pet) {
-        this.pet = pet;
+    public SlotBuilder withPet(String name) {
+        this.pet = typicalPetTracker.getPet(new Name(name));;
```
Comment (shenghaoc) [inline src/test/java/seedu/address/testutil/slot/SlotBuilder.java:46]: For consistency all inputs to withXXX should mimic user input, i.e. be String.
Anyway it makes sense to extract out Name declaration and getPet now that we have hardcoded typicalPets, which we always use, into Slotbuilder.

**Review by Zhu-Ruicong — COMMENTED** [review]
LGTM

**Pull request merged**